### ISSUE TITLE
Allow setting the host (0.0.0.0) on which the API listens 

### ISSUE THREAD

**Issue by beeman** [kin-labs/kinetic#364]
We need to be able to set the `HOST` env var that determines on which hosts our API listens, defaulting to `127.0.0.1` but giving the developer the option to change it (e.g. use 0.0.0.0)

We can mimic the behavior of the `PORT` env var.

To do:

- [ ] add support for `HOST` env var
- [ ] add it to `validation-schema` and `configuration` 
- [ ] make sure that env var has a default value of `127.0.0.1`
- [ ] add it to the `ApiConfigDataAccessService` as the `host` getter
- [ ] use `config.host` in the `app.listen` method, just like `port`.

If you can create an issue for the aiove and add that as a separate PR, we can rebase this once it lands.

_Fix for issue pointed out by @RamirezAlex in https://github.com/kin-labs/kinetic/pull/358#discussion_r954189960_
